src/vs/platform/extensionManagement/common/extensionManagement.ts

```diff
@@ -291,7 +291,7 @@ export const IExtensionTipsService = createDecorator<IExtensionTipsService>('ext
 
 export interface IExtensionTipsService {
 	_serviceBrand: any;
-	getRecommendations(): string[];
+	getRecommendations(): { [type: string]: string[]; };
```

Comment:
One small suggestion - Why not the API returns an object with `file` and `exe` as properties? So that it will be transparent what it is returning.

Reply:
You mean array of objects like
```
[
  {
      type: "file",
      extensions: [
         ....,
        ....
      ]
  },
  {
      type: "exe",
      extensions: [
         ....,
        ....
      ]
  }
]
```

Reply:
@sandy081  Actually, the only reason I wanted them separate was to be able to add telemetry which i can do from inside `getRecommendations()`

So reverted back to sending `string[]`